### PR TITLE
Implement fmt::Display descriptions for FsBlockDbError

### LINKED ISSUE #742: Implement `std::fmt::Display` for `FsBlockDbError`
We need to implement a comprehensive set of errors for the FFI layer.

````
  Err(e) => Err(format_err!("Error while scanning blocks: {}", e)),
     |                                                                          ^ `FsBlockDbError` cannot be formatted with the default formatter
     |
     = help: the trait `std::fmt::Display` is not implemented for `FsBlockDbError`
````

### PR DESCRIPTION
**Pull request by pacu**
Closes #742

This is mainly for FFI Display

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
utACK modulo Clippy fixes.

**Comment by pacu** [comment]
Clippy fixed. Thanks!

**Comment by pacu** [comment]
Rustfmt and clippy don't get along. I opened a ticket around this issue

https://github.com/zcash/librustzcash/issues/750

File: zcash_client_sqlite/src/lib.rs [hunk 1]
```diff
@@ -951,6 +952,21 @@ impl BlockSource for FsBlockDb {
...
+impl std::fmt::Display for FsBlockDbError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            FsBlockDbError::Fs(io_error) => write!(f, "FsBlockDbError::Fs({})", io_error),
+            FsBlockDbError::Db(e) => write!(f, "FsBlockDbError::Db({})", e),
+            FsBlockDbError::Protobuf(e) => write!(f, "FsBlockDbError::Protobuf({})", e),
+            FsBlockDbError::MissingBlockPath(_) => write!(f, "FsBlockDbError::MissingBlockPath"),
+            FsBlockDbError::InvalidBlockstoreRoot(_) => write!(f, "FsBlockDbError::InvalidBlockstoreRoot"),
+            FsBlockDbError::InvalidBlockPath(_) => write!(f, "FsBlockDbError::InvalidBlockPath"),
+            FsBlockDbError::CorruptedData(e) => write!(f, "FsBlockDbError::CorruptedData({})", e),
```
Comment (str4d) [inline zcash_client_sqlite/src/lib.rs:965]: These are not suitable descriptions for a `Display` impl, as they should be user-meaningful (vs `Debug` impls for which these would be fine). See https://github.com/zcash/librustzcash/blob/125d2bc3d58c543ad58eb53b6c65c0d616688f52/zcash_client_sqlite/src/error.rs#L98-L128 for a (terse but sufficient) example of an error `Display` impl.

File: zcash_client_backend/src/welding_rig.rs [hunk 2]
```diff
@@ -203,7 +203,8 @@ pub(crate) fn scan_block_with_runner<
     tree: &mut CommitmentTree<Node>,
     existing_witnesses: &mut [&mut IncrementalWitness<Node>],
     mut batch_runner: Option<&mut TaggedBatchRunner<P, K::Scope, T>>,
-) -> Vec<WalletTx<K::Nf>> {
+) -> Vec<WalletTx<K::Nf>> 
+{
```
Comment (str4d) [inline zcash_client_backend/src/welding_rig.rs:207]: Have `rustfmt` take precedence here and leave in the clippy warnings.

File: zcash_client_sqlite/src/lib.rs [hunk 3]
```diff
@@ -523,7 +523,8 @@ impl<'a, P: consensus::Parameters> WalletWrite for DataConnStmtCache<'a, P> {
         &mut self,
         block: &PrunedBlock,
         updated_witnesses: &[(Self::NoteRef, IncrementalWitness<Node>)],
-    ) -> Result<Vec<(Self::NoteRef, IncrementalWitness<Node>)>, Self::Error> {
+    ) -> Result<Vec<(Self::NoteRef, IncrementalWitness<Node>)>, Self::Error> 
+    {
```
Comment (str4d) [inline zcash_client_sqlite/src/lib.rs:527]: Ditto.

**Review by str4d — CHANGES_REQUESTED** [review]

File: zcash_client_sqlite/src/lib.rs [hunk 4]
```diff
@@ -951,6 +951,21 @@ impl BlockSource for FsBlockDb {
...
 }
 
+#[cfg(feature = "unstable")]
+impl std::fmt::Display for FsBlockDbError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            FsBlockDbError::Fs(io_error) => write!(f, "Failed to access the file system: {}", io_error),
+            FsBlockDbError::Db(e) => write!(f, "There was a problem with the sqlite db: {}", e),
+            FsBlockDbError::Protobuf(e) => write!(f, "Failed to parse protobuf-encoded record: {}", e),
+            FsBlockDbError::MissingBlockPath(_) => write!(f, "A path to a cached block was provided but it can't be found"),
```
Comment (str4d) [inline zcash_client_sqlite/src/lib.rs:961]:
```suggestion
            FsBlockDbError::MissingBlockPath(block_path) => write!(f, "CompactBlock file expected but not found at {}", block_path.display()),
```

File: zcash_client_sqlite/src/lib.rs [hunk 5]
```diff
@@ -951,6 +951,21 @@ impl BlockSource for FsBlockDb {
...
+            FsBlockDbError::InvalidBlockstoreRoot(_) => write!(f, "The block storage root provided is invalid"),
```
Comment (str4d) [inline zcash_client_sqlite/src/lib.rs:962]:
```suggestion
            FsBlockDbError::InvalidBlockstoreRoot(fsblockdb_root) => write!(f, "The block storage root {} is not a directory", fsblockdb_root.display()),
```

File: zcash_client_sqlite/src/lib.rs [hunk 6]
```diff
@@ -951,6 +951,21 @@ impl BlockSource for FsBlockDb {
...
+            FsBlockDbError::InvalidBlockPath(_) => write!(f, "The block path provided is invalid."),
```
Comment (str4d) [inline zcash_client_sqlite/src/lib.rs:963]:
```suggestion
            FsBlockDbError::InvalidBlockPath(block_path) => write!(f, "CompactBlock path {} is not a file", block_path.display()),
```

**Review by str4d — CHANGES_REQUESTED** [review]
The code also needs a `cargo fmt`.

**Comment by pacu** [comment]
I fixed `cargo fmt` issues and added @str4d 's

**Review by str4d — APPROVED** [review]
utACK

**Pull request merged**